File: sfml demos/C++/exercise 04/SFML/Gamepad.cpp
#include "Gamepad.h"



Gamepad::Gamepad(const int index)
{
	this->index = index;
	if (this->isConnected(this->index))
	{
		cout << "Gamepad connected\n" << endl;
		cout << getID() << endl;
		active = true;
		buttons = getButtonCount(this->index);
		hasZ = hasAxis(this->index, Z);
	}
	else
	{
		cout << "Gamepad disconnected\n" << endl;
		active = false;
	}
}


Gamepad::~Gamepad()
{
}

string Gamepad::getID()
{
	Identification id = getIdentification(this->index);	
	return 
		"\n\tVendor ID: " + to_string(id.vendorId) +
		"\n\tJoystick use: " + id.name.toAnsiString() +
		"\n\tProduct ID: " + to_string(id.productId) +
		"\n\tButton count: " + to_string(buttons) +
		"\n\tHas z-axis: " + to_string(hasZ);
}

int Gamepad::getButtonPressed()
{
	for (int i = 0; i < buttons; i++)
		if (isButtonPressed(index, i))
			return i;
}

File: C#/exercise 06/sfml.net/src/Gamepad.cs
using SFML.Window;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using static SFML.Window.Joystick;

namespace sfml.net.src
{
    class Gamepad
    {
        #region Fields

        private uint index;
        public uint Index { get { return index; } }

        private bool isActive = false;
        private bool hasZ = false;
        private bool move = false;
        private uint buttonCount = 0;
        private int deadzone = 15;

        public float ButtonCount
        {
            get { return Joystick.GetButtonCount(this.index); }
        }
        public Identification Identification
        {
            get { return Joystick.GetIdentification(this.index); }
        }
        public bool IsConnected
        {
            get { return Joystick.IsConnected(this.index); }
        }


        public uint GetButtonPressed
        {
            get
            {
                for (uint i = 0; i < this.buttonCount; i++)
                    if (Joystick.IsButtonPressed(this.index, i))
                        return i;
                return
                    this.buttonCount + 1;
            }
        }

        #endregion


        #region Public 

        public Gamepad(uint index)
        {
            this.index = index;
            if (Joystick.IsConnected(this.index))
            {
                Console.WriteLine(String.Format("Gamepad Connected\nID:{0}\n", Joystick.GetIdentification(this.index)));
                this.hasZ = Joystick.HasAxis(this.index, Joystick.Axis.Z);
                this.isActive = true;
                this.buttonCount = Joystick.GetButtonCount(this.index);
            }
            else
            {
                Console.WriteLine("No gamepad connected.");
                this.isActive = false;
            }
        }

        public float GetAxisPosition(Axis axis)
        {
            return Joystick.GetAxisPosition(this.index, axis);
        }

        public bool HasAxis(Axis axis)
        {
            return Joystick.HasAxis(this.index, axis);
        }

        public bool IsButtonPressed(uint button)
        {
            return Joystick.IsButtonPressed(this.index, button);
        }

        

        #endregion
    }
}
